staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -773,7 +775,11 @@ func (c *Cacher) GetList(ctx context.Context, key string, opts storage.ListOptio
 		// minimal resource version, simply forward the request to storage.
 		return c.storage.GetList(ctx, key, opts, listObj)
 	}
-	if resourceVersion == "" && utilfeature.DefaultFeatureGate.Enabled(features.ConsistentListFromCache) {
+	requestWatchProgressSupported, err := etcdfeature.DefaultFeatureSupportChecker.Supports(storage.RequestWatchProgress)
```

Comment:
We can arbitrarily complicate it - doing that per resource, periodically checking because etcd can be upgraded underneath, etc.). We made a conscious decision in the KEP that we want this to be simple.